src/checkpoint/dispatch/vrt/serializer_registry.h

```diff
@@ -109,18 +116,33 @@ TypeIdx const Type<ObjT, SerializerT>::idx =
   Registrar<ObjT, SerializerT>().index;
 
 template <typename ObjT>
-inline auto getObjIdx(TypeIdx han) {
+inline auto& getObjIdxRef(TypeIdx han) {
   checkpointAssert(
-    han < getRegistry<ObjT>().size(),
+    han < static_cast<TypeIdx>(getRegistry<ObjT>().size()),
     "Missing type idx in registry. A derived type was not registered with "
     "checkpoint for a given serializer"
   );
-  return std::get<1>(getRegistry<ObjT>().at(han));
+  return getRegistry<ObjT>().at(han);
 }
 
-template <typename ObjT, typename SerializerT>
-inline TypeIdx makeObjIdx() {
-  return Type<ObjT, SerializerT>::idx;
+template <typename ObjT>
+inline auto getObjIdx(TypeIdx han) {
+  auto elm = getObjIdxRef<ObjT>(han);
+  return std::get<2>(elm);
+}
+
+template <typename DerivedT>
+inline std::function<void(void*, DerivedT&)> getBaseIdx(TypeIdx base_idx) {
```

Comment:
I don't see this actually being used.

If it is, I don't understand at all what it's doing.

Reply:
It see it's used in a later commit. I still don't really understand what it's doing.